src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1498 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-bus.h"
+
+#include "bus-get-properties.h"
+#include "bus-label.h"
+#include "bus-log-control-api.h"
+#include "bus-polkit.h"
+#include "common-signal.h"
+#include "discover-image.h"
+#include "env-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "json.h"
+#include "main-func.h"
+#include "memfd-util.h"
+#include "missing_capability.h"
+#include "mkdir-label.h"
+#include "os-util.h"
+#include "process-util.h"
+#include "service-util.h"
+#include "signal-util.h"
+#include "socket-util.h"
+#include "string-table.h"
+#include "user-util.h"
+
+typedef struct Manager {
+        sd_event *event;
+        sd_bus *bus;
+
+        Hashmap *targets;
+
+        uint32_t current_job_id;
+        Hashmap *jobs;
+
+        Hashmap *polkit_registry;
+
+        int notify_fd;
```

Comment:
in new code please just make use of sd_event_source_set_io_fd_own() and pass ownership of the fd to notify_event_source, so that it closes the fd on its own, and we don't need to store the fd separately.